src/System.CommandLine/Argument.cs

```diff
@@ -105,14 +103,15 @@ private protected override string DefaultName
             }
         }
 
-        internal List<Action<ArgumentResult>> Validators => _validators ??= new ();
+        internal IReadOnlyList<Action<ArgumentResult>> Validators
+            => _validators is null ? Array.Empty<Action<ArgumentResult>>() : _validators;
 
         /// <summary>
         /// Adds a custom validator to the argument. Validators can be used
         /// to provide custom errors based on user input.
         /// </summary>
         /// <param name="validate">The action to validate the parsed argument.</param>
-        public void AddValidator(Action<ArgumentResult> validate) => Validators.Add(validate);
+        public void AddValidator(Action<ArgumentResult> validate) => (_validators ??= new()).Add(validate);
```

Comment:
There are other places that call Validator.Add, which would now add the Action to the Array.Empty<Action<ArgumentResult>>.

https://github.com/dotnet/command-line-api/blob/350a618ab44932c568ae2392d7a571281baed59a/src/System.CommandLine/Argument%7BT%7D.cs#L215

https://github.com/dotnet/command-line-api/blob/350a618ab44932c568ae2392d7a571281baed59a/src/System.CommandLine/OptionValidation.cs#L21